src/compiler/checker.ts

```diff
@@ -11860,15 +11869,23 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     }
 
     function reportCircularityError(symbol: Symbol) {
-        const declaration = symbol.valueDeclaration as VariableLikeDeclaration;
+        const declaration = symbol.valueDeclaration;
         // Check if variable has type annotation that circularly references the variable itself
-        if (getEffectiveTypeAnnotationNode(declaration)) {
-            error(symbol.valueDeclaration, Diagnostics._0_is_referenced_directly_or_indirectly_in_its_own_type_annotation, symbolToString(symbol));
-            return errorType;
+        if (declaration) {
+            if (getEffectiveTypeAnnotationNode(declaration)) {
+                error(symbol.valueDeclaration, Diagnostics._0_is_referenced_directly_or_indirectly_in_its_own_type_annotation, symbolToString(symbol));
+                return errorType;
+            }
+            // Check if variable has initializer that circularly references the variable itself
+            if (noImplicitAny && (declaration.kind !== SyntaxKind.Parameter || (declaration as HasInitializer).initializer)) {
+                error(symbol.valueDeclaration, Diagnostics._0_implicitly_has_type_any_because_it_does_not_have_a_type_annotation_and_is_referenced_directly_or_indirectly_in_its_own_initializer, symbolToString(symbol));
+            }
         }
-        // Check if variable has initializer that circularly references the variable itself
-        if (noImplicitAny && (declaration.kind !== SyntaxKind.Parameter || (declaration as HasInitializer).initializer)) {
-            error(symbol.valueDeclaration, Diagnostics._0_implicitly_has_type_any_because_it_does_not_have_a_type_annotation_and_is_referenced_directly_or_indirectly_in_its_own_initializer, symbolToString(symbol));
+        else if (symbol.flags & SymbolFlags.Alias) {
+            const symbolImport = symbol.declarations?.find(getAnyImportSyntax);
+            if (symbolImport) {
+                error(symbolImport, Diagnostics.Circular_definition_of_import_alias_0, symbolToString(symbol));
+            }

```

Comment:
I'm not quite sure `getAnyImportSyntax` is the right helper; that seems to be something specific to visibility rules.

Reading the other place where `Circular_definition_of_import_alias_0` is used (`resolveAlias`), probabably `getDeclarationOfAliasSymbol` is a good location to put the symbol.

```suggestion
            const node = getDeclarationOfAliasSymbol(symbol);
            if (node) {
                error(node, Diagnostics.Circular_definition_of_import_alias_0, symbolToString(symbol));
            }

```

Reply:
Are imports always the last declaration? `getDeclarationOfAliasSymbol` returns the last declaration that is true for `isAliasSymbolDeclaration`, which from https://github.com/microsoft/TypeScript/blob/a25321afce6ca2ead8332f7e0d6f923ff239c671/src/compiler/types.ts#L5725
seems like it is similar to only checking for `SymbolFlags.Alias` and may not be specific enough to make sure that the declaration is an import?

Reply:
No idea; but `getDeclarationOfAliasSymbol` also only checks `& SymbolKind.Symbol` and then uses this helper to issue the same error.